FindAllShortestPaths.cpp

```diff
@@ -0,0 +1,198 @@
+#include <algorithm>
+#include <iostream>
+#include <vector>
+#include <limits>
+#include <stack>
+#include <list>
+
+typedef std::vector<std::list<std::pair<int, int>>> adjacency_list; // std::pair<vertex, weight to that edge>
+
+// this function deletes the vertices that have more length than `ShortestDistance`
+void DeleteNotShortestPaths(std::vector<std::vector<std::pair<int, int>>>& Paths, int vertex, int ShortestDistance)
+{
+    // if vector is not empty and to your vertex path is longer than `ShortestDistance`, them we clear all that vector and start to fill it with `ShortestDistance`
+    if (!Paths[vertex].empty() && Paths[vertex].begin()->second > ShortestDistance)
+    {
+        Paths[vertex].clear();
+    }
+}
+
+// this Dijkstra is modified an dwill retunr matrix that has information about all shortest paths
+std::vector<std::vector<std::pair<int, int>>> ModifiedDijkstra(adjacency_list Graph, int start)
+{
+    std::vector<int> distance(Graph.size(), std::numeric_limits<int>::max());
+    std::vector<char> used(Graph.size(), 'f'); // never use vector of bools
```

Comment:
What is this

Reply:
For Dijkstra's algorithm, I took a vector that assigns the maximum possible value to all vertices. After that I create a vector char where I could write true, false and not 'f', 't'. Now I have already fixed it

Reply:
You have forgotten to make the vector `bool` type